### ISSUE TITLE
Textpattern 4.8.6 release flight plan

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1673]
We are targeting Textpattern 4.8.6 for release in Q2 2021 to address some snags, and provide further refinements.

* [4.8.6 branch](https://github.com/textpattern/textpattern/tree/4.8.6)

Please update this post and subsequent replies with tasks etc as needed, assigning yourself or others where appropriate.

**Repo Issues**

- [ ] textpattern/textpattern 
  * 4.8.6 milestone: [open issues](https://github.com/textpattern/textpattern/milestone/19) & [closed issues](https://github.com/textpattern/textpattern/milestone/19?closed=1)
- [ ] textpattern/textpacks
  * 4.8.6 milestone: [open issues](https://github.com/textpattern/textpacks/issues?q=is%3Aopen+is%3Aissue+milestone%3Av4.8.6) & [closed issues](https://github.com/textpattern/textpacks/issues?q=is%3Aclosed+is%3Aissue+milestone%3Av4.8.6)

## Personnel

* Release pilot: tbc
* Release co-pilot: tbc

## Preflight

- [ ] preflight
  - [x] latest translations (@philwareham)
  - [x] latest pophelp (@philwareham)
  - [x] draft .com blog post [ID 440] (release pilot, or assigned person)
  - [x] write change history (everyone)
  - [x] tag public/admin themes with release tag (@philwareham)
  - [x] tag Textpacks and Pophelps with release tag (@philwareham)

## Take Off

- [ ] take off
  - [x] package up for GitHub and .com (release pilot)
  - [x] update `latest.zip` and `latest.tar.gz` in server config (@petecooper)
  - [x] update language pack links in server config (@petecooper)
  - [ ] update demo site (@petecooper)
  - [ ] publish .com blog post (release pilot, or assigned person)
  - [ ] forum announcement (release pilot, or assigned person)
  - [ ] forum feedback (release pilot, or assigned person)

## Inflight service

- [ ] inflight service
  - [x] close milestone (release pilot, or assigned person)
  - [x] update `TXPversion` on `rpc.textpattern.com` (release pilot, or assigned person)
  - [ ] update https://en.wikipedia.org/wiki/Template:Textpattern_version (@petecooper or @philwareham)
  - [ ] socials (@petecooper or @philwareham)
  - [ ] repin https://github.com/textpattern/textpattern/issues/968

**Comment by petecooper** [comment]
Reserved for important notes.

**Comment by petecooper** [comment]
Three commits to merge into `4.8.6` when it's ready to go:

- [ ] https://github.com/textpattern/textpattern/commit/ef2a430c7127a6c6c5c800b93a37a74f56cff810
- [ ] https://github.com/textpattern/textpattern/commit/3bef7bfd7a492700dc6db52987c3a9334b03e56c
- [ ] https://github.com/textpattern/textpattern/commit/5ad20cb43bc3a631b58f40943bc1f4c19807a41f

**Comment by Bloke** [comment]
Oki doke. Branch created, cherry-picked above commits, and bi-merged dev<>4.8.6.

**Comment by petecooper** [comment]
First boarding call for anything else that needs to be included in 4.8.6. I think we should be able to turn this one around pretty smoothly. I'll make a start on the .com announcement today, got some spare time this evening.

**Comment by petecooper** [comment]
I'll have little to no web access from Apr 26 to May 7 as the office is being packed up and moved, so I won't be around reliably for that time. If we're close to release for 4.8.6, can we shoot for this week or second week of May, perhaps?

**Comment by petecooper** [comment]
Bumping this.

Are we close?

**Comment by Bloke** [comment]
I've added https://github.com/textpattern/textpacks/issues/304 to the pre-requisites in the OP and I'm happy that #1653 is done. A couple of recent commits (today) that change the article tag could do with some grilling but otherwise, looking good for take-off.

**Comment by philwareham** [comment]
I’ll look at the Textpack change on Monday. Nothing else from my side.

**Comment by bloatware** [comment]
I've got no more planned commits for 4.8.6, they all come from users requests flow. Personally, I would prefer that clicking 'Assign sections' links on Themes tab preselect only the sections from *this* theme, but it is of no importance.

**Comment by Bloke** [comment]
> I would prefer that clicking 'Assign sections' links on Themes tab preselect only the sections from _this_ theme.

Me too.

**Comment by bloatware** [comment]
> Me too.

Just ask :-) Feel free to revert if not pleased.

**Comment by philwareham** [comment]
Textpacks and Pophelps are up to date. So no further things needed from me for 4.8.6 (apart from tagging the various repos when I'm told to). Note I'm on holiday next week so would be good to get this out before end of Friday latest otherwise I won't be able to help until the week commencing 7th June.

**Comment by Bloke** [comment]
Let's aim for release Thursday or Friday then, barring any last-minute setbacks.

**Comment by petecooper** [comment]
I'm around for release assistance as needed, Friday evening is clear currently.

**Comment by philwareham** [comment]
I can do that now. Will let you know when done.

**Comment by Bloke** [comment]
Super minor thing that doesn't need attention now, but are the setup strings under CrowdIn control? I just happened to spot that the de.ini has a missing space after the `</strong>` tag in:

~~~
tables_exist="Die Tabellen in der Datenbank <strong>{dbname}</strong>existieren ...
~~~

If I change it locally in the release is that going to screw up CrowdIn sync in future? If so, I'll leave it and we can push it from the top down sometime in future.

**Comment by philwareham** [comment]
Please fix in this textpattern repo, 4.8.6 branch - I will sort Crowdin and the actual Textpack repo in a few minutes (just tagging theme releases at the moment). Ta.

**Comment by Bloke** [comment]
Will do, ta.

**Comment by philwareham** [comment]
OK, I've tagged themes and textpacks/pophelp repos for 4.8.6. My pre-release stuff is done. I can update WikiPedia after release as that's the only other thing on my list above. Good luck everyone!

**Comment by Bloke** [comment]
Woot! Thank you, Sir.

**Comment by petecooper** [comment]
Confirming availability from now until bedtime.

@Bloke – please holler as you see fit.

**Comment by Bloke** [comment]
Thank you.

Waiting on confirmation of [this test](https://github.com/textpattern/textpattern/commit/c351ee26aef35459eee2d5f59dd9dc69ce91955b#commitcomment-51434435) in case it's PEBKAC on my part or something that needs fixing. If you (or anyone) want to give it a whirl on a demo site or something and see if you can replicate it, that'd be ace.

I'll be on the road for the next few hours but should be able to pick up where I left off later. I've branched locally, just waiting on the above results, then I can commit "This is 4.8.6" and go from there with the remainder of the string-pulling.

**Comment by bloatware** [comment]
Should be fixed now, sorry for the hassle.

**Comment by Bloke** [comment]
Everyone: Please welcome Textpattern 4.8.6. Another great release with some fabulous goodies woven into its fabric. Thank you so much to everyone who has contributed and, yet again, the awesomeness of you all for keeping this wonderful ball of code rolling. **high fives everyone I can reach**

I've slogged through pretty much the entire release process. Things outstanding:

* Bounce the .com nginx server.
* Delete the GitHub 4.8.6 branch - it had dangling refs again and wouldn't let me do it from command line. Gotta figure out why one day.
* Publish the .com article - I think it's good to go but I won't have time in the morning.
* Do the publicising on the forum/socials.
* Fix anything in the .com articles (e.g. Get Started etc) to point to the 4.8.6 release blog article.
* Documentation...

I've made a 4.8.7 branch for when we roll any PHP 8.1 stuff in, so keep that in mind as we forge ahead with 4.9.0.

Onward and upwards...

**Comment by petecooper** [comment]
Possible breakage on the auto-installer, though it could be PEBKAC honestly.

4.8.6 auto-installer throws an error relating to database user permissions:

```
$ sudo bash /var/www/sites/textpattern.co/release-demo/staging/release-demo_build.sh
mysql: [Warning] Using a password on the command line interface can be insecure.
[OK]	Connected
[ERROR]	Insufficient database privileges for release_demo@127.0.0.1. Requires: SELECT, CREATE, ALTER, INSERT, UPDATE, DELETE, DROP, INDEX, LOCK TABLES.
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'release_demo.txp_users' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
mysql: [Warning] Using a password on the command line interface can be insecure.
```

The user in question has all those privileges. I have deleted and recreated both the database and the user numerous times in testing.

The same build script works A-OK with 4.8.5, literally all I do on the demo site is bump the branch name to check out, then run the build script, and voila.

Ideas and advice more than welcome!

**Comment by Bloke** [comment]
Weird. How come it's been building fine for 4.8.6-dev demo all this time and as soon as it's released it breaks? That's frustrating. I hope I haven't screwed anything up.

**Comment by petecooper** [comment]
Rats. `dev-demo` is also failing. I've been focussed elsewhere recently, when was `dev-demo` working properly according to browser history?

**Comment by Bloke** [comment]
I just ran it locally and it installed fine. I then changed permissions for the user and it errored out as expected. So something's going on in the demo server environment.

You shouldn't need it, but there's a new option in the config file 'create' which will auto-create the database if it doesn't exist:

``` json
{
    "site": {
        "language_code": "en",
        "public_url": "",
        "theme": "",
        "public_theme": "",
        "datadir": ""
    },
    "database": {
        "user": "db-user",
        "password": "db-password",
        "host": "127.0.0.1",
        "db_name": "",
        "table_prefix": "",
        "create": "1"
    },
    "user": {
        "full_name": "Some Person",
        "email": "email@example.com",
        "login_name": "Manager12345",
        "password": "1ab1u8287ecwie"
    }
}
```

**Comment by petecooper** [comment]
PEBKAC. Good.

I will flatten and repave after lunch as time permits. Should not be a blocker (unless you want it to be) for the rest of promotion etc but please comment out "demo site has been updated" in the announcement post.

Will get to this ASAP today.

**Comment by Bloke** [comment]
Aye but [this is a problem](https://github.com/textpattern/textpattern/issues/1689). Something's stuffed up and it's not taking PFX into account on the Users panel. I'm just leaving now and won't be at a terminal for the foreseeable of the day.

So if anyone has any time to investigate, that would be most appreciated. Just install a new Txp with a table prefix, stick debugging on and then visit the Users panel. BOOM.

If I'm guessing I'd say it's the change to `getCount()` or `getThings()` yesterday that I should have tested further. Or in the way I'm building the search in the revamped Users panel. But it might be in the Search/Filter.php (although if it is, why does it work on other panels, hmmm).

Thoughts welcome. And then what: keep quiet about 4.8.6 and silently re-release, or skip it and go to 4.8.7.

**Comment by petecooper** [comment]
>skip it and go to 4.8.7.

^^ This (my 2c).

**Comment by petecooper** [comment]
Commencing server rebuild to investigate https://github.com/textpattern/textpattern/issues/1673#issuecomment-850736495 in more detail.

**Comment by petecooper** [comment]
>I just ran it locally and it installed fine. I then changed permissions for the user and it errored out as expected. So something's going on in the demo server environment.

Out of interest – MySQL 5.something or 8.something? I can reliably break 4.8.6 auto-installer with MySQL, whereas 4.8.5 works as expected. Server nuked and repaved, checked and double checked. Gonna be afk for a few hours, hope to be back by 8pm or so.

**Comment by Bloke** [comment]
Ah it was mysql 5.6.40. I'll run it again with 8.0.0 and see what's going on, thanks. Swear I tested that. I must be going mad.

**Comment by petecooper** [comment]
@Bloke – I'll email you.

**Comment by petecooper** [comment]
Closing in favour of https://github.com/textpattern/textpattern/issues/1690

Please stand by for release.
